openmdao/docs/openmdao_book/features/model_visualization/reports_system.ipynb

```diff
@@ -0,0 +1,247 @@
+    "\n",
+    "### Controlling reporting using the Problem reports argument \n",
+    "The user can also control what reports get run using the `reports` argument to `Problem`.\n",
+    "\n",
+    "If `reports` is the default of `_UNDEFINED`, the `OPENMDAO_REPORTS` environment variable \n",
+    "determines reporting.\n",
+    "\n",
+    "If the value for `reports` is a Boolean, it will enable/disable all reports on that `Problem`. A value of `None` is equivalent to `False`.\n",
+    "\n",
+    "Otherwise, `reports` may be a comma-delimted string of the names of the reports to run for that Problem."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting the directory for the reports\n",
+    "\n",
+    "If the user wishes to have the reports directory placed into a different directory other than the default current working\n",
+    "directory, the user can set the environment variable, `OPENMDAO_REPORTS_DIR` to the name of that directory. The\n",
+    "directory does not have to exist beforehand. It will be created by the reporting system. As an example, if `OPENMDAO_REPORTS_DIR` is set to \"my_reports\", then, if the `Problem` name is `problem1`, then the reports will be written to the \"my_reports/problem1\" directory.\n",
+    "\n",
+    "## Support for subproblems\n",
+    "\n",
+    "The reporting system supports subproblems. For example, if a model has two Problems and they have the default names of\n",
+    "`problem1` and `problem2`, the reporting system will put the reports into the folders, `problem1_reports` and\n",
+    "`problem2_reports`.\n",
+    "\n",
+    "## Adding user defined reports\n",
+    "\n",
+    "The user can register a user defined report with the reporting system using the `register_report` function.\n",
+    "\n",
+    "```{eval-rst}\n",
+    "    .. autofunction:: openmdao.utils.reports_system.register_report\n",
+    "        :noindex:\n",
+    "```\n",
+    "\n",
+    "Here is a script to show how this is done. Note that the function that is called to do the reporting must have\n",
+    "as its first argument, the class name of the OpenMDAO object owning the method where the report will be run. Currently, the only options are `Problem` and `Driver`.\n",
+    "\n",
+    "After this script is run, there will be a file named `user_report.txt` in the directory `problem1_reports`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "remove-input",
+     "remove-output"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ['OPENMDAO_REPORTS'] = '1'  # Need to run reports in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import pathlib\n",
+    "from openmdao.utils.reports_system import register_report, get_reports_dir\n",
```

Comment:
`register_reports` and `get_reports_dir` should be imported from `openmdao.api` in the documentation, to simplify things for users.